### ISSUE TITLE
misc-macro-parentheses misses some uses of arguments

### ISSUE THREAD

**Issue by LegalizeAdulthood** [llvm/llvm-project#26646]
|  |  |
| --- | --- |
| Bugzilla Link | [26272](https://llvm.org/bz26272) |
| Version | unspecified |
| OS | Linux |

## Extended Description 
Example:
```
clanger> cat /tmp/b.cpp
#define FOO(b0, b1) \
  b0=b1; \
  b1++
~/dev/id/iterated-dynamics
clanger> ~/dev/build/bin/clang-tidy -checks='-*,misc-macro-parentheses' /tmp/b.cpp -- -std=c++11
1 warning generated.
/tmp/b.cpp:3:3: warning: macro argument should be enclosed in parentheses [misc-macro-parentheses]
  b1++
  ^
  ( )
```
clang-tidy complains about the 2nd usage of b1, but doesn't complain about the usage of b0 or the first usage of b1.  The expected changed output should be:
```
#define FOO(b0, b1) \
  (b0)=(b1); \
  (b1)++
```
